awd/cli.py

```diff
@@ -157,40 +157,71 @@ def deposit(
-            ctx.obj["doi_table"]
-        )
+        try:
+            doi_items = dynamodb_client.retrieve_doi_items_from_database(
+                ctx.obj["doi_table"]
+            )
+        except ClientError as e:
+            logger.error(f"Table read failed: {e.response['Error']['Message']}")
+            exit
         for doi in dois:
             if doi_to_be_added(doi, doi_items):
-                dynamodb_client.add_doi_item_to_database(ctx.obj["doi_table"], doi)
+                try:
+                    dynamodb_client.add_doi_item_to_database(ctx.obj["doi_table"], doi)
+                except ClientError as e:
+                    logger.error(
+                        f"Table update failed: {e.response['Error']['Message']}"
+                    )
+
             elif doi_to_be_retried(doi, doi_items) is False:
                 continue
-            dynamodb_client.update_doi_item_status_in_database(
-                ctx.obj["doi_table"], doi, Status.PROCESSING.value
-            )
-            dynamodb_client.update_doi_item_attempts_in_database(
-                ctx.obj["doi_table"], doi
-            )
-            crossref_work_record = crossref.get_work_record_from_doi(metadata_url, doi)
-            if crossref.is_valid_response(doi, crossref_work_record) is False:
+            try:
                 dynamodb_client.update_doi_item_status_in_database(
-                    ctx.obj["doi_table"], doi, Status.FAILED.value
+                    ctx.obj["doi_table"], doi, Status.PROCESSING.value
+                )
+            except ClientError as e:
+                logger.error(f"Table update failed: {e.response['Error']['Message']}")
+            try:
+                dynamodb_client.update_doi_item_attempts_in_database(
+                    ctx.obj["doi_table"], doi
                 )
+            except ClientError as e:
+                logger.error(f"Table update failed: {e.response['Error']['Message']}")
+            crossref_work_record = crossref.get_work_record_from_doi(metadata_url, doi)
+            if crossref.is_valid_response(doi, crossref_work_record) is False:
+                try:
+                    dynamodb_client.update_doi_item_status_in_database(
+                        ctx.obj["doi_table"], doi, Status.FAILED.value
+                    )
+                except ClientError as e:
+                    logger.error(
+                        f"Table update failed: {e.response['Error']['Message']}"
+                    )
                 continue
             value_dict = crossref.get_metadata_extract_from(crossref_work_record)
             metadata = crossref.create_dspace_metadata_from_dict(
                 value_dict, "config/metadata_mapping.json"
             )
             if crossref.is_valid_dspace_metadata(metadata) is False:
-                dynamodb_client.update_doi_item_status_in_database(
-                    ctx.obj["doi_table"], doi, Status.FAILED.value
-                )
+                try:
```

Comment:
Is there a way to refactor this block of statements (as they appear multiple times)?

Reply:
Refactored, the try/except wasn't changing the flow of the overall process so I moved it to the methods